broker-daemon/state-machines/plugins/state-machine-events.js

```diff
@@ -17,14 +17,25 @@ class StateMachineEvents extends StateMachinePlugin {
   }
 
   /**
-   * Adds an event emition observer to all transition calls to allow for event passing
+   * Adds event emition observer to all transition calls to allow for event passing
    * to external workers (BlockOrderWorker)
    */
   get observers () {
     const plugin = this
 
     return {
       /**
+       * Emit an event before a transition (prefixed by 'before:')
+       * e.g. stateMachine.on('before:create')
+       * @param  {Object} lifecycle
+       * @return {void}
+       */
+      onBeforeTransition: function (lifecycle) {
```

Comment:
@treygriffith did you want to add this in this PR? It doesn't look like we use it anywhere

Reply:
@dannypaz I just pushed a change that does use it - I accidentally left it out, thanks for the catch.